src/Temporalio/Workflows/VersioningIntent.cs

```diff
@@ -0,0 +1,45 @@
+namespace Temporalio.Workflows
+{
+    /// <summary>
+    /// Indicates whether the user intends certain commands to be run on a compatible worker Build
+    /// Id version or not.
+    ///
+    /// Where this type is accepted optionally, an unset value indicates that the SDK should choose the
+    /// most sensible default behavior for the type of command, accounting for whether the command will
+    /// be run on the same task queue as the current worker.
+    /// </summary>
+    public abstract record VersioningIntent
```

Comment:
I think this should be a C# enum

Reply:
Same as other comment, I think we should use the C# enum feature even if your internal switch statement loses this compile time check.